# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1658562659 -7200
# Parent  9a9fcd839e815a7944e5b9010186d91bd2367b90
Branding 6: Clean up About box, switch off the updater, fix donation links.
* * *
Misc: Add ! to helpus.end, to see get 'Make a donation!'

diff --git a/mail/base/content/aboutDialog.xhtml b/mail/base/content/aboutDialog.xhtml
--- a/mail/base/content/aboutDialog.xhtml
+++ b/mail/base/content/aboutDialog.xhtml
@@ -37,19 +37,19 @@
 
   <script src="chrome://messenger/content/aboutDialog.js"/>
 #ifdef MOZ_UPDATER
   <script src="chrome://messenger/content/aboutDialog-appUpdater.js"/>
 #endif
 
   <vbox id="aboutDialogContainer">
     <hbox id="clientBox">
-      <vbox id="leftBox" flex="1"/>
+      <vbox id="leftBox" flex="0"/>
       <vbox id="rightBox" flex="1">
-        <hbox align="baseline">
+        <hbox align="baseline" style="margin-top: 10px;">
 #expand   <label id="version">__MOZ_APP_VERSION_DISPLAY__</label>
           <label is="text-link" id="releasenotes" hidden="true">&releaseNotes.link;</label>
         </hbox>
 
         <label id="distribution" class="text-blurb"/>
         <label id="distributionId" class="text-blurb"/>
 
         <vbox id="detailsBox">
@@ -150,46 +150,31 @@
 #endif
           </vbox>
 
 #ifdef MOZ_UPDATER
           <description class="text-blurb" id="currentChannelText">
             &channel.description.start;<label id="currentChannel"/>&channel.description.end;
           </description>
 #endif
+          <!-- experimental is switched off via JS for release builds -->
           <vbox id="experimental" hidden="true">
-            <description class="text-blurb" id="warningDesc">
-              &warningDesc.version;
-#ifdef MOZ_TELEMETRY_ON_BY_DEFAULT
-              &warningDesc.telemetryDesc;
-#endif
-            </description>
-            <description class="text-blurb" id="communityExperimentalDesc">
-              &community.exp.start;<label class="text-link" onclick="openLink('https://www.mozilla.org/');">&community.exp.mozillaLink;</label>&community.exp.middle;<label class="text-link" onclick="openAboutTab('about:credits');">&community.exp.creditsLink;</label>&community.exp.end;
-            </description>
           </vbox>
+          <description class="text-blurb" style="margin-bottom: 0; font-size: 2em;">
+            <label class="text-link" onclick="openLink('https://www.betterbird.eu/');">www.betterbird.eu</label>
+          </description>
+          <!-- communityDesc is switched off via JS for Daily builds -->
           <description class="text-blurb" id="communityDesc">
-            &community.start2;<label class="text-link" onclick="openLink('https://www.mozilla.org/');">&community.mozillaLink;</label>&community.middle2;<label class="text-link" onclick="openAboutTab('about:credits');">&community.creditsLink;</label>&community.end3;
           </description>
           <description class="text-blurb" id="contributeDesc">
-            &helpus.start;<label class="text-link" onclick="openLink('https://give.thunderbird.net/?utm_source=thunderbird-client&amp;utm_medium=referral&amp;utm_content=about-dialog');">&helpus.donateLink;</label>&helpus.middle;<label class="text-link" onclick="openLink('https://www.thunderbird.net/get-involved/')">&helpus.getInvolvedLink;</label>&helpus.end;
+            &helpus.start;<label class="text-link" onclick="openLink('https://www.betterbird.eu/donate/');">&helpus.donateLink;</label>&helpus.end;
           </description>
         </vbox>
       </vbox>
     </hbox>
-    <vbox id="bottomBox">
-      <hbox pack="center">
-        <label class="text-link bottom-link" onclick="openAboutTab('about:license');">&bottomLinks.license;</label>
-        <label class="text-link bottom-link" onclick="openAboutTab('about:rights');">&bottomLinks.rights;</label>
-        <label class="text-link bottom-link" onclick="openLink('https://www.mozilla.org/privacy/thunderbird/');">&bottomLinks.privacy;</label>
-      </hbox>
-      <description id="trademark">
-        <label class="trademark-label">&trademarkInfo.part1;</label>
-      </description>
-    </vbox>
   </vbox>
 
   <keyset id="mainKeyset">
     <key keycode="VK_ESCAPE" oncommand="window.close();"/>
 #ifdef XP_MACOSX
     <key id="key_close" modifiers="accel" key="&cmdCloseMac.commandKey;"
          oncommand="window.close();"/>
 #endif
diff --git a/mail/base/content/helpMenu.inc.xhtml b/mail/base/content/helpMenu.inc.xhtml
--- a/mail/base/content/helpMenu.inc.xhtml
+++ b/mail/base/content/helpMenu.inc.xhtml
@@ -5,44 +5,24 @@
 <menu id="helpMenu"
 #ifdef XP_WIN
             label="&helpMenuWin.label;" accesskey="&helpMenuWin.accesskey;"
 #else
             label="&helpMenu.label;" accesskey="&helpMenu.accesskey;"
 #endif
             onpopupshowing="buildHelpMenu();">
         <menupopup id="menu_HelpPopup">
-          <menuitem id="menu_openHelp"
-                    label="&productHelp.label;"
-#ifndef XP_MACOSX
-                    accesskey="&productHelp.accesskey;"
-#endif
-                    key="key_openHelp"
-                    oncommand="openSupportURL();"/>
-          <menuitem id="menu_openTour"
-                    label="&helpShowTour2.label;"
-                    accesskey="&helpShowTour2.accesskey;"
-                    oncommand="openLinkText(event, 'tourURL');"/>
           <menuitem id="menu_keyboardShortcuts"
                     label="&helpKeyboardShortcuts.label;"
                     accesskey="&helpKeyboardShortcuts.accesskey;"
                     oncommand="openLinkText(event, 'keyboardShortcutsURL');"/>
-          <menuseparator/>
-          <menuitem id="getInvolved"
-                    label="&helpGetInvolvedPage.label;"
-                    accesskey="&helpGetInvolvedPage.accesskey;"
-                    oncommand="openLinkText(event, 'getInvolvedURL');"/>
           <menuitem id="donationsPage"
                     label="&helpDonationsPage.label;"
                     accesskey="&helpDonationsPage.accesskey;"
                     oncommand="openLinkText(event, 'donateURL');"/>
-          <menuitem id="feedbackPage"
-                    label="&helpFeedbackPage.label;"
-                    accesskey="&helpFeedbackPage.accesskey;"
-                    oncommand="openLinkText(event, 'feedbackURL');"/>
           <menuseparator id="functionsSeparator"/>
           <menuitem id="helpTroubleshootMode"
                     data-l10n-id="menu-help-enter-troubleshoot-mode"
                     oncommand="safeModeRestart();"/>
           <menuitem id="aboutsupport_open"
                     data-l10n-id="menu-help-more-troubleshooting-info"
                     oncommand="openAboutSupport();"/>
 #ifndef XP_MACOSX
diff --git a/mail/base/content/utilityOverlay.js b/mail/base/content/utilityOverlay.js
--- a/mail/base/content/utilityOverlay.js
+++ b/mail/base/content/utilityOverlay.js
@@ -204,20 +204,17 @@ function openLinkText(event, what) {
   switch (what) {
     case "getInvolvedURL":
       openUILink("https://www.thunderbird.net/get-involved/", event);
       break;
     case "keyboardShortcutsURL":
       openUILink("https://support.mozilla.org/kb/keyboard-shortcuts/", event);
       break;
     case "donateURL":
-      openUILink(
-        "https://give.thunderbird.net/?utm_source=thunderbird-client&utm_medium=referral&utm_content=help-menu",
-        event
-      );
+      openUILink("https://www.betterbird.eu/donate/", event);
       break;
     case "tourURL":
       openUILink("https://www.thunderbird.net/features/", event);
       break;
     case "feedbackURL":
       openUILink("https://connect.mozilla.org/", event);
       break;
   }
diff --git a/mail/branding/betterbird/content/aboutDialog.css b/mail/branding/betterbird/content/aboutDialog.css
--- a/mail/branding/betterbird/content/aboutDialog.css
+++ b/mail/branding/betterbird/content/aboutDialog.css
@@ -7,22 +7,22 @@
 }
 
 #leftBox {
   background-image: url("chrome://branding/content/about-logo.svg");
   background-repeat: no-repeat;
   /* min-width and min-height create room for the logo */
   min-width: 210px;
   min-height: 210px;
-  margin-top: 20px;
   margin-inline-start: 30px;
 }
 
 #rightBox {
   background-size: 260px auto;
+  margin-top: 40px;
 }
 
 #updateDeck > hbox > label:not([class="text-link"]) {
   opacity: 0.6;
 }
 
 #trademark {
   font-size: xx-small;
diff --git a/mail/components/about-support/content/aboutSupport.js b/mail/components/about-support/content/aboutSupport.js
--- a/mail/components/about-support/content/aboutSupport.js
+++ b/mail/components/about-support/content/aboutSupport.js
@@ -121,16 +121,17 @@ function toFluentID(str) {
     .toLowerCase();
 }
 
 // Each property in this object corresponds to a property in Troubleshoot.jsm's
 // snapshot data.  Each function is passed its property's corresponding data,
 // and it's the function's job to update the page with it.
 var snapshotFormatters = {
   async application(data) {
+    // This comes form Services.appinfo.name likely via MOZ_APP_BASENAME.
     $("application-box").textContent = "Betterbird";
     $("useragent-box").textContent = data.userAgent;
     $("os-box").textContent = data.osVersion;
     if (data.osTheme) {
       $("os-theme-box").textContent = data.osTheme;
     } else {
       $("os-theme-row").hidden = true;
     }
diff --git a/mail/components/customizableui/content/panelUI.inc.xhtml b/mail/components/customizableui/content/panelUI.inc.xhtml
--- a/mail/components/customizableui/content/panelUI.inc.xhtml
+++ b/mail/components/customizableui/content/panelUI.inc.xhtml
@@ -1772,38 +1772,24 @@
       </vbox>
     </panelview>
 
     <!-- Help -->
     <panelview id="appMenu-helpView"
                title="&helpMenuWin.label;"
                class="PanelUI-subView">
       <vbox class="panel-subview-body">
-        <toolbarbutton id="appmenu_openHelp"
-                       class="subviewbutton subviewbutton-iconic"
-                       label="&productHelp.label;"
-                       key="key_openHelp"
-                       oncommand="openSupportURL();"/>
         <toolbarbutton id="menu_keyboardShortcuts"
                        class="subviewbutton subviewbutton-iconic"
                        label="&helpKeyboardShortcuts.label;"
                        oncommand="openLinkText(event, 'keyboardShortcutsURL');"/>
-        <toolbarseparator/>
-        <toolbarbutton id="getInvolved"
-                       class="subviewbutton subviewbutton-iconic"
-                       label="&helpGetInvolvedPage.label;"
-                       oncommand="openLinkText(event, 'getInvolvedURL');"/>
         <toolbarbutton id="donationsPage"
                        class="subviewbutton subviewbutton-iconic"
                        label="&helpDonationsPage.label;"
                        oncommand="openLinkText(event, 'donateURL');"/>
-        <toolbarbutton id="feedbackPage"
-                       class="subviewbutton subviewbutton-iconic"
-                       label="&helpFeedbackPage.label;"
-                       oncommand="openLinkText(event, 'feedbackURL');"/>
         <toolbarseparator/>
         <toolbarbutton id="appmenu_troubleshootMode"
                        class="subviewbutton subviewbutton-iconic"
                        data-l10n-id="appmenu-help-enter-troubleshoot-mode"
                        oncommand="safeModeRestart();"/>
         <toolbarbutton id="appmenu_troubleshootingInfo"
                        class="subviewbutton subviewbutton-iconic"
                        data-l10n-id="appmenu-help-more-troubleshooting-info"
diff --git a/mail/confvars.sh b/mail/confvars.sh
--- a/mail/confvars.sh
+++ b/mail/confvars.sh
@@ -1,13 +1,15 @@
 #! /bin/sh
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+# We need to keep MOZ_APP_NAME thunderbird, otherwise all the code that checks
+# AppConstants.MOZ_APP_NAME won't work any more, and there is such code in M-C!
 MOZ_APP_NAME=thunderbird
 
 if test "$OS_ARCH" = "WINNT"; then
   if ! test "$HAVE_64BIT_BUILD"; then
     MOZ_VERIFY_MAR_SIGNATURE=1
   fi
 fi
 
diff --git a/mail/locales/en-US/chrome/messenger/aboutDialog.dtd b/mail/locales/en-US/chrome/messenger/aboutDialog.dtd
--- a/mail/locales/en-US/chrome/messenger/aboutDialog.dtd
+++ b/mail/locales/en-US/chrome/messenger/aboutDialog.dtd
@@ -42,17 +42,17 @@
 <!ENTITY community.end3             " working together to keep the Web open, public and accessible to all.">
 
 <!ENTITY helpus.start               "Want to help? ">
 <!-- LOCALIZATION NOTE (helpus.donateLink): This is a link title that links to the thunderbird donation page -->
 <!ENTITY helpus.donateLink          "Make a donation">
 <!ENTITY helpus.middle              " or ">
 <!-- LOCALIZATION NOTE (helpus.getInvolvedLink): This is a link title that links to https://www.thunderbird.net/get-involved/. -->
 <!ENTITY helpus.getInvolvedLink     "get involved!">
-<!ENTITY helpus.end                 "">
+<!ENTITY helpus.end                 "!">
 
 <!ENTITY releaseNotes.link          "Release notes">
 
 <!-- LOCALIZATION NOTE (bottomLinks.license): This is a link title that links to about:license. -->
 <!ENTITY bottomLinks.license        "Licensing Information">
 
 <!-- LOCALIZATION NOTE (bottomLinks.rights): This is a link title that links to about:rights. -->
 <!ENTITY bottomLinks.rights         "End-User Rights">
diff --git a/mail/moz.configure b/mail/moz.configure
--- a/mail/moz.configure
+++ b/mail/moz.configure
@@ -2,16 +2,17 @@
 # vim: set filetype=python:
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 set_config("MOZ_THUNDERBIRD", True)
 set_define("MOZ_THUNDERBIRD", True)
 
+# We can't set this to Betterbird or else Thunderbird profiles won't be recognised.
 imply_option("MOZ_APP_BASENAME", "Thunderbird")
 set_config("MOZ_APPUPDATE_HOST", "aus.thunderbird.net")
 
 imply_option("--enable-default-browser-agent", False)
 imply_option("MOZ_REQUIRE_SIGNING", False)
 
 
 @depends(target_is_windows, target_is_linux)
